libusb/descriptor.c

```diff
@@ -384,7 +393,10 @@ static int parse_configuration(struct libusb_context *ctx,
 	buffer += config->bLength;
 	size -= config->bLength;
 
-	for (i = 0; i < config->bNumInterfaces; i++) {
+	uint8_t stashedNumInterfaces = config->bNumInterfaces;
+	config->bNumInterfaces = 0;
+
+	for (; config->bNumInterfaces < stashedNumInterfaces; config->bNumInterfaces++) {
```

Comment:
Same as previous: ensures that the interfaces are cleared in the case of an error.